<launch>
  <!-- 
       Default GUI to true for local Gazebo client. Allows override 
       to set disable GUI for RoboMaker. See part0.launch. 
  -->
  <arg name="use_gui" default="true"/>


  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" value="$(find exercises)/robot_chase/worlds/chase_world.sdf"/>
    <arg name="paused" value="true"/>
    <arg name="use_sim_time" value="true"/>
    <arg name="gui" value="$(arg use_gui)"/>
    <arg name="headless" value="false"/>
    <arg name="debug" value="false"/>
  </include>


<!-- https://answers.ros.org/question/197701/multiple-turtlebots-simulation-in-gazebo-and-visualization-in-rviz/ --> 



<group ns = "cop_0">
  <param name="robot_description" command="$(find xacro)/xacro --inorder $(find turtlebot3_description)/urdf/turtlebot3_burger.urdf.xacro" />
<node pkg="gazebo_ros" type="spawn_model" name="spawn_urdf"  args="-urdf -model cop_0 -x -7 -y -7 -z 0 -Y 0 -param robot_description" />
</group>


<group ns = "cop_1">
  <param name="robot_description" command="$(find xacro)/xacro --inorder $(find turtlebot3_description)/urdf/turtlebot3_burger.urdf.xacro" />
<node pkg="gazebo_ros" type="spawn_model" name="spawn_urdf"  args="-urdf -model cop_1 -x -6 -y -7 -z 0 -Y 0 -param robot_description" />
</group>


<group ns = "cop_2">
  <param name="robot_description" command="$(find xacro)/xacro --inorder $(find turtlebot3_description)/urdf/turtlebot3_burger.urdf.xacro" />
<node pkg="gazebo_ros" type="spawn_model" name="spawn_urdf"  args="-urdf -model cop_2 -x -5 -y -7 -z 0 -Y 0 -param robot_description" />
</group>


<group ns = "robber_0">
  <param name="robot_description" command="$(find xacro)/xacro --inorder $(find turtlebot3_description)/urdf/turtlebot3_burger.urdf.xacro" />
<node pkg="gazebo_ros" type="spawn_model" name="spawn_urdf"  args="-urdf -model robber_0 -x 7 -y 7 -z 0 -Y 1.3 -param robot_description" />
</group>


<group ns = "robber_1">
  <param name="robot_description" command="$(find xacro)/xacro --inorder $(find turtlebot3_description)/urdf/turtlebot3_burger.urdf.xacro" />
<node pkg="gazebo_ros" type="spawn_model" name="spawn_urdf"  args="-urdf -model robber_1 -x -7 -y 7 -z 0 -Y 1.3 -param robot_description" />
</group>


<group ns = "robber_2">
  <param name="robot_description" command="$(find xacro)/xacro --inorder $(find turtlebot3_description)/urdf/turtlebot3_burger.urdf.xacro" />
<node pkg="gazebo_ros" type="spawn_model" name="spawn_urdf"  args="-urdf -model robber_2 -x 7 -y -7 -z 0 -Y 1.3 -param robot_description" />
</group>



</launch>
